tests/main/interfaces-personal-files/task.yaml

```diff
@@ -38,7 +72,43 @@ execute: |
     echo "When the interface is connected"
     snap connect test-snapd-sh:personal-files
 
-    echo "Then the snap is able to access all the files and dirs in $HOME"
+    echo "There is a single \"ensure-dir\" kind of user mount entry for \$HOME/.local/share"
+    test -e /var/lib/snapd/mount/snap.test-snapd-sh.user-fstab
+    test "$(wc -l </var/lib/snapd/mount/snap.test-snapd-sh.user-fstab)" -eq 1
+    # shellcheck disable=SC2016
+    MATCH 'none \$HOME/.missing none x-snapd.kind=ensure-dir,x-snapd.must-exist-dir=\$HOME 0 0' < /var/lib/snapd/mount/snap.test-snapd-sh.user-fstab
+
+    echo "Cannot abuse snap-confine to create $TEST_USER_HOME/.missing owned by user $TEST_USER2"
+    # This command is based on what snap-confine contructs for running a non-classic app with non-root user
+    cmd="SNAP_INSTANCE_NAME=test-snapd-sh SNAP_REVISION=x1 SNAP_REAL_HOME=$TEST_USER_HOME /usr/lib/snapd/snap-confine snap.test-snapd-sh.with-personal-files-plug /usr/lib/snapd/snap-exec test-snapd-sh.with-personal-files-plug -c exit"
+    if tests.session -u $TEST_USER2 exec sh -c "$cmd" > call.error 2>&1; then
+        echo 'Expected error: "snap-update-ns failed with code 1"'
+        exit 1
+    fi
+    # shellcheck disable=SC2016
+    MATCH 'cannot update snap namespace: cannot expand mount entry \(none \$HOME/.missing none x-snapd.kind=ensure-dir,x-snapd.must-exist-dir=\$HOME 0 0\): cannot use invalid home directory '"\"$TEST_USER_HOME\": permission denied" < call.error
+    MATCH "snap-update-ns failed with code 1" < call.error
+
+    echo "Cannot abuse snap-confine to create $ROOT_OWNED_DIR/.missing owned by user $TEST_USER2"
+    # This command is based on what snap-confine contructs for running a non-classic app with non-root user
+    cmd="SNAP_INSTANCE_NAME=test-snapd-sh SNAP_REVISION=x1 SNAP_REAL_HOME=$ROOT_OWNED_DIR /usr/lib/snapd/snap-confine snap.test-snapd-sh.with-personal-files-plug /usr/lib/snapd/snap-exec test-snapd-sh.with-personal-files-plug -c exit"
+    tests.session -u $TEST_USER2 exec sh -c "$cmd" 2> call.error
+    MATCH "update.go:85: cannot change mount namespace according to change mount \(none $ROOT_OWNED_DIR/.missing none x-snapd.kind=ensure-dir,x-snapd.must-exist-dir=/etc 0 0\): cannot create directory \"$ROOT_OWNED_DIR/.missing\": permission denied" < call.error
```

Comment:
Small nitpick, maybe it's better to avoid line numbers in the matching pattern

Reply:
Thanks, fixed